File: api/v2/Server/Routes/QualityRoutes.py
from api.v2.Contract.ApiResponse import ApiResponse
from api.v2.Server.CoreApiServer import CoreApiServer


class QualityRoutes:
    """集中注册质量规则相关 HTTP 路由。"""

    UPDATE_META_PATH: str = "/api/v2/quality/rules/update-meta"
    SAVE_ENTRIES_PATH: str = "/api/v2/quality/rules/save-entries"
    IMPORT_RULES_PATH: str = "/api/v2/quality/rules/import"
    EXPORT_RULES_PATH: str = "/api/v2/quality/rules/export"
    RULE_PRESETS_PATH: str = "/api/v2/quality/rules/presets"
    RULE_PRESET_READ_PATH: str = "/api/v2/quality/rules/presets/read"
    RULE_PRESET_SAVE_PATH: str = "/api/v2/quality/rules/presets/save"
    RULE_PRESET_RENAME_PATH: str = "/api/v2/quality/rules/presets/rename"
    RULE_PRESET_DELETE_PATH: str = "/api/v2/quality/rules/presets/delete"
    STATISTICS_PATH: str = "/api/v2/quality/rules/statistics"
    PROMPT_TEMPLATE_PATH: str = "/api/v2/quality/prompts/template"
    PROMPT_SAVE_PATH: str = "/api/v2/quality/prompts/save"
    PROMPT_IMPORT_PATH: str = "/api/v2/quality/prompts/import"
    PROMPT_EXPORT_PATH: str = "/api/v2/quality/prompts/export"
    PROMPT_PRESETS_PATH: str = "/api/v2/quality/prompts/presets"
    PROMPT_PRESET_READ_PATH: str = "/api/v2/quality/prompts/presets/read"
    PROMPT_PRESET_SAVE_PATH: str = "/api/v2/quality/prompts/presets/save"
    PROMPT_PRESET_RENAME_PATH: str = "/api/v2/quality/prompts/presets/rename"
    PROMPT_PRESET_DELETE_PATH: str = "/api/v2/quality/prompts/presets/delete"

    @classmethod
    def register(cls, core_api_server: CoreApiServer, quality_rule_app_service) -> None:
        core_api_server.add_json_route(
            "POST",
            cls.UPDATE_META_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.update_rule_meta(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.SAVE_ENTRIES_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.save_rule_entries(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.IMPORT_RULES_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.import_rules(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.EXPORT_RULES_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.export_rules(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.RULE_PRESETS_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.list_rule_presets(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.RULE_PRESET_READ_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.read_rule_preset(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.RULE_PRESET_SAVE_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.save_rule_preset(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.RULE_PRESET_RENAME_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.rename_rule_preset(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.RULE_PRESET_DELETE_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.delete_rule_preset(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.STATISTICS_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.build_rule_statistics(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_TEMPLATE_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.get_prompt_template(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_SAVE_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.save_prompt(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_IMPORT_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.import_prompt(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_EXPORT_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.export_prompt(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_PRESETS_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.list_prompt_presets(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_PRESET_READ_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.read_prompt_preset(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_PRESET_SAVE_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.save_prompt_preset(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_PRESET_RENAME_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.rename_prompt_preset(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.PROMPT_PRESET_DELETE_PATH,
            lambda request: ApiResponse(
                ok=True, data=quality_rule_app_service.delete_prompt_preset(request)
            ),
        )

File: src/core/api/routes/quality-routes.ts
import type { ApiRouteContext } from "./route-types";

export function register_quality_routes(context: ApiRouteContext): void {
  const query = context.services.workbench.query;
  const statistics = context.services.quality.statistics;
  const quality = context.services.quality.service;

  context.postJson("/api/project/query/quality-statistics", (body) => statistics.read(body));
  context.postJson("/api/project/query/quality-rule", (body) => query.read_quality_rule_view(body));
  context.postJson("/api/project/query/prompt", (body) => query.read_prompt_view(body));
  context.postJson("/api/quality/rules/save-entries", (body) => quality.save_rule_entries(body));
  context.postJson("/api/quality/rules/update-meta", (body) => quality.update_rule_meta(body));
  context.postJson("/api/quality/rules/import", (body) => quality.import_rules(body));
  context.postJson("/api/quality/rules/export", (body) => quality.export_rules(body));
  context.postJson("/api/quality/rules/presets", (body) => quality.list_rule_presets(body));
  context.postJson("/api/quality/rules/presets/read", (body) => quality.read_rule_preset(body));
  context.postJson("/api/quality/rules/presets/save", (body) => quality.save_rule_preset(body));
  context.postJson("/api/quality/rules/presets/rename", (body) => quality.rename_rule_preset(body));
  context.postJson("/api/quality/rules/presets/delete", (body) => quality.delete_rule_preset(body));
  context.postJson("/api/quality/prompts/template", (body) => quality.get_prompt_template(body));
  context.postJson("/api/quality/prompts/save", (body) => quality.save_prompt(body));
  context.postJson("/api/quality/prompts/import", (body) => quality.read_prompt_import_text(body));
  context.postJson("/api/quality/prompts/export", (body) => quality.export_prompt(body));
  context.postJson("/api/quality/prompts/presets", (body) => quality.list_prompt_presets(body));
  context.postJson("/api/quality/prompts/presets/read", (body) => quality.read_prompt_preset(body));
  context.postJson("/api/quality/prompts/presets/save", (body) => quality.save_prompt_preset(body));
  context.postJson("/api/quality/prompts/presets/rename", (body) =>
    quality.rename_prompt_preset(body),
  );
  context.postJson("/api/quality/prompts/presets/delete", (body) =>
    quality.delete_prompt_preset(body),
  );
}
